core/src/main/java/org/infinispan/util/logging/Log.java

```diff
@@ -2140,4 +2140,12 @@ CacheConfigurationException storeConfiguredHasBothReadAndWriteOnly(String storeC
    @LogMessage(level = WARN)
    @Message(value = "Index path '%s' is not absolute and global state is disabled, will use a dir relative to the current working directory.", id = 626)
    void indexRelativeWorkingDir(String path);
+
+   @LogMessage(level = WARN)
+   @Message(value = "Exception completing partial completed transaction %s. Retrying later.", id = 627)
+   void failedPartitionHandlingTxCompletion(GlobalTransaction globalTransaction, @Cause Throwable t);
+
+   @LogMessage(level = WARN)
+   @Message(value = "Another partition or topology changed for while completing partial completed transaction %s. Retrying later.", id = 628)
```

Comment:
I don't think this should be a warning. Debug might work because txs are completed on a single thread, so it will only be logged once per topology change, but trace is best IMO.

Reply:
why trace is better? I don't think it is that often the topology change after the stable topology is installed.

Reply:
@pruivo because `SuspectException` is not something that users should ever see a scary WARN message about: the whole point of Infinispan is to hide the fact that another node has stopped/failure from the application.

Also, my initial comment was wrong: even though the commit commands are sent on a single thread, the targets are always computed based on the stable topology, and the procedure doesn't stop after a `SuspectException`. So if there's a topology change in the middle, the same warning is logged for all the transactions that follow.

Reply:
The `SuspectException` is a bad example because it doesn't log this particular log message.
I'm actually assuming that topology changes are a rare event, especially after installing a stable topology.
But if you insist, I can open a JIRA to remove all the warnings.

Reply:
> The SuspectException is a bad example because it doesn't log this particular log message.

@pruivo are you sure? When commitNodes != null, the rpc uses `MapResponseCollector.validOnly()`, which throws `SuspectException` on `CacheNotFoundResponse`

It doesn't really matter though, because you also log a warning on `CacheNotFoundResponse` later.

I'd keep this warning for exceptions other han `SuspectException`, maybe even make it an ERROR because it should never happen.

I'd also keep the `topologyChangedPartitionHandlingTxCompletion` warning, but log it once per topology change, not once per transaction.

Reply:
@danberindei what I mean is the SuspectException is logged in another place :) 
I've created this PR https://github.com/infinispan/infinispan/pull/9214 
can you have a look when you have time?